kibana-reports/server/routes/utils/savedSearchReportHelper.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { v1 as uuidv1 } from 'uuid';
+import {
+  buildQuery,
+  convertToCSV,
+  getEsData,
+  getSelectedFields,
+  metaData,
+} from './dataReportHelpers';
+import {
+  IClusterClient,
+  IScopedClusterClient,
+} from '../../../../../src/core/server';
+
+/**
+ * Specify how long scroll context should be maintained for scrolled search
+ */
+const scrollTimeout = '1m';
+
+export async function createSavedSearchReport(
+  report: any,
+  client: IClusterClient | IScopedClusterClient
+) {
+  await populateMetaData(client, report);
+  const data = await generateCsvData(
+    client,
+    report.report_definition.report_params.core_params.limit
+  );
+
+  const timeCreated = new Date().toISOString();
+  const fileName = getFileName() + '.csv';
```

Comment:
If a future `TODO` item is to extend support for other file formats like `xlsx`, should we consider extracting the file format to an input that is passed in from client-side as opposed to hard-coding it from the server-side?

Reply:
Good catch! I hardcode this accidentally. Will use the format in input. Thanks!